rpc/dialer.go

```diff
@@ -254,7 +254,8 @@ func dialDirectGRPC(ctx context.Context, address string, dOpts dialOptions, logg
 	// Use SOCKS proxy from environment as gRPC proxy dialer. Do not use
 	// if trying to connect to a local address.
 	if proxyAddr := os.Getenv(SocksProxyEnvVar); proxyAddr != "" &&
-		!(strings.HasPrefix(address, "[::]") || strings.HasPrefix(address, "localhost")) {
+		!(strings.HasPrefix(address, "[::]") || strings.HasPrefix(address, "localhost") ||
+			strings.HasPrefix(address, "unix")) {
```

Comment:
basically a `unix` address means we are dialing locally right?

kind feels like this could either pull out a function - `validSOCKSAddress` or something - which could be a good place add docs about various inclusions/exclusions as needed. anyway, won't block, just a thought

Reply:
works for me!